src/index/hgraph_index.cpp

```diff
@@ -169,13 +179,14 @@ HGraphIndex::knn_search(const DatasetPtr& query,
         while (search_result.size() > k) {
             search_result.pop();
         }
+        const auto& allocator = this->common_param_.allocator_.get();
```

Comment:
minor: We can add an field `shared_ptr<SafeAllocator> allocator_` to hold `common_param_.allocator` ?